### PR TITLE
feat: Migrate Registry to spell

### PR DESCRIPTION
**Pull request by gurinderu**

### CODE REVIEW

File: crates/system-services/Cargo.toml [hunk 1]
```diff
@@ -8,7 +8,7 @@ edition = "2021"
 [dependencies]
 aqua-ipfs-distro = "=0.5.14"
 decider-distro = "=0.4.15"
-registry-distro = "=0.8.5"
+registry-distro = { git = "https://github.com/fluencelabs/registry.git", branch = "feature/net-256"}
```
Comment (kmd-fl) [inline crates/system-services/Cargo.toml:11]: Maybe just publish the registry-distro and use the update version?

**Review by kmd-fl — APPROVED** [review]
Fine by me. Need to update to the latest system-service-deployer, though. Would be nice to merge this PR asap

File: crates/server-config/src/defaults.rs [hunk 2]
```diff
@@ -224,6 +224,26 @@ pub fn default_decider_spell_period_sec() -> u32 {
     120
 }
 
+// 60 minutes
+pub fn default_registry_spell_period_sec() -> u32 {
+    3600
```
Comment (folex) [inline crates/server-config/src/defaults.rs:229]: what is this period for? others are clearly named, but this one i don't understand

Comment (gurinderu) [reply crates/server-config/src/defaults.rs:229]: it is a period for running a spell

Comment (folex) [reply crates/server-config/src/defaults.rs:229]: it's not really clear that these 4 values must be connected. ie, `default_registry_spell_period_sec` must always be a smallest common denominator.

File: crates/system-services/src/deployer.rs [hunk 3]
```diff
@@ -149,12 +150,10 @@ impl Deployer {
     }
 
     async fn deploy_registry(&self) -> eyre::Result<()> {
-        let (registry_distro, registry_spell_distros) = Self::get_registry_distro();
+        let (registry_distro, registry_spell_distro) =
+            Self::get_registry_distro(self.config.registry.clone());
         let _deployed = self.deploy_service_common(registry_distro)?;
```
Comment (folex) [inline crates/system-services/src/deployer.rs:155]:
```suggestion
        self.deploy_service_common(registry_distro)?;
```

File: crates/system-services/src/deployer.rs [hunk 4]
```diff
@@ -198,29 +197,30 @@ impl Deployer {
...
-        (distro, spells_distro)
+
+        let registry_config = registry_distro::RegistryConfig {
+            expired_interval: config.expired_period_sec,
+            renew_interval: config.renew_period_sec,
+            replicate_interval: config.replicate_period_sec,
+        };
+        let spell_distro = registry_distro::registry_spell(registry_config);
+        let mut trigger_config = TriggerConfig::default();
+        trigger_config.clock.start_sec = 1;
```
Comment (folex) [inline crates/system-services/src/deployer.rs:214]: looks like this deserves a comment. Why not keep it default `0`?

**Review by folex — APPROVED** [review]

File: crates/server-config/src/defaults.rs (hunk 2 above)
Comment (gurinderu) [reply crates/server-config/src/defaults.rs:229]: Done

File: crates/system-services/src/deployer.rs (hunk 4 above)
Comment (gurinderu) [reply crates/system-services/src/deployer.rs:214]: Used the same value as for decider

Comment (folex) [reply crates/system-services/src/deployer.rs:214]: why?

Comment (gurinderu) [reply crates/system-services/src/deployer.rs:214]: why not)

Comment (gurinderu) [reply crates/system-services/src/deployer.rs:214]: Added comment

**Pull request merged**